kratos/solving_strategies/strategies/explicit_solving_strategy.h

```diff
@@ -0,0 +1,719 @@
+     * @param RebuildLevel The flag to set if the DOF set is rebuild or not
+     */
+    explicit ExplicitSolvingStrategy(
+        ModelPart &rModelPart,
+        bool MoveMeshFlag = false,
+        int RebuildLevel = 0)
+        : mrModelPart(rModelPart)
+    {
+        SetMoveMeshFlag(MoveMeshFlag);
+        SetRebuildLevel(RebuildLevel);
+        mpExplicitBuilderAndSolver = Kratos::make_unique<ExplicitBuilderAndSolver<TSparseSpace, TDenseSpace>>();
+    }
+
+    /** Copy constructor.
+     */
+    ExplicitSolvingStrategy(const ExplicitSolvingStrategy &Other) = delete;
+
+    /** Destructor.
+     */
+    virtual ~ExplicitSolvingStrategy() = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * @brief Operation to predict the solution ... if it is not called a trivial predictor is used in which the
+     * values of the solution step of interest are assumed equal to the old values
+     */
+    virtual void Predict()
+    {
+        if (!GetInitializeWasPerformedFlag()) {
+            Initialize();
+        }
+
+        if (!GetInitializeSolutionStepWasPerformedFlag()) {
+            InitializeSolutionStep();
+        }
+    }
+
+    /**
+     * @brief Initialization of member variables and prior operations
+     */
+    virtual void Initialize()
+    {
+        if (!GetInitializeWasPerformedFlag()) {
+            // Initialize elements, conditions and constraints
+            InitializeContainer(GetModelPart().Elements());
+            InitializeContainer(GetModelPart().Conditions());
+            InitializeContainer(GetModelPart().MasterSlaveConstraints());
+
+            // Set the explicit DOFs rebuild level
+            if (mRebuildLevel != 0) {
+                mpExplicitBuilderAndSolver->SetResetDofSetFlag(true);
+            }
+
+            // If the mesh is updated at each step, we require to accordingly update the lumped mass at each step
+            if (mMoveMeshFlag) {
+                mpExplicitBuilderAndSolver->SetResetLumpedMassVectorFlag(true);
+            }
+
+            // Call the explicit builder and solver initialize (Set up DOF set and lumped mass vector)
+            mpExplicitBuilderAndSolver->Initialize(mrModelPart);
+
+            // Initialize the solution values
+            InitializeDofSetValues();
+
+            // Set the mInitializeWasPerformed flag
+            mInitializeWasPerformed = true;
+        }
+    }
+
+    /**
+     * @brief The problem of interest is solved.
+     * @details
+     * {
+     * This function calls sequentially: Initialize(), InitializeSolutionStep(), Predict(), SolveSolutionStep() and FinalizeSolutionStep().
+     * All those functions can otherwise be called separately.
+     * }
+     */
+    virtual double Solve()
+    {
+        Initialize();
+        InitializeSolutionStep();
+        Predict();
+        SolveSolutionStep();
+        FinalizeSolutionStep();
+
+        return 0.0;
+    }
```

Comment:
```suggestion
```

better not add this as it usually gives problems due to misuse

=> better call individually

Reply:
OK. Removed.